### PR TITLE
Fill in detail for V2 registry api specification

### PR DESCRIPTION
**Pull request by stevvooe**
These changes fill in a lot of missing detail in the API endpoint definitions. Please the generated `SPEC.md` file for the results. This file will be submitted to docker core for review, as a companion to docker/docker#9784.

Note that this PR contains part of the content from #7, which should be merged before this.

cc @icecrime @jfrazelle @dmp42 @dmcgowan @jlhawn

### CODE REVIEW

**Comment by dmp42** [comment]
LGTM

File: api/v2/descriptors.go [hunk 1]
```diff
@@ -417,17 +529,33 @@ var routeDescriptors = []RouteDescriptor{
...
+									},
+									contentLengthZeroHeader,
+								},
 							},
 						},
 						Failures: []ResponseDescriptor{
 							{
-								StatusCode: http.StatusBadRequest,
+								Name:        "Invalid Manifest",
+								Description: "The recieved manifest was invalid in some way, as described by the error codes. The client should resolve the issue and retry the request.",
```
Comment (dmcgowan) [inline api/v2/descriptors.go]: Typo

File: api/v2/descriptors.go [hunk 2]
```diff
@@ -629,7 +890,8 @@ var routeDescriptors = []RouteDescriptor{
 						},
 						Successes: []ResponseDescriptor{
 							{
-								StatusCode: http.StatusCreated,
+								Description: "The blob has been created in the registry and is available the provided location.",
```
Comment (dmcgowan) [inline api/v2/descriptors.go]: Missing an "at"

**Comment by dmcgowan** [comment]
LGTM

**Comment by stevvooe** [comment]
Superseded by #20.

**Pull request closed without merging**